specs/unmanage_cluster.adoc

```diff
@@ -0,0 +1,357 @@
+un-manage cluster and after success import cluster. UI should maintain both the
+tasks details for detailing in UI
+
+
+=== Alternatives
+
+None
+
+=== Data model impact
+
+* Change the fields `import_job_id` and `import_status` as `current_job` and
+`status` respectively for cluster entity
+
+* The same fields would be updated with appropriate details while import and
+un-manage flows on cluster
+
+* The field `current_job` would maintain a dict containing `status`, `job_name`
+and `job_id` for currently running job on cluster
+
+* The field `status` would maintain values like `importing`, `unmanaging`,
+`syncing` or `unknown` at a time. This maintains any flows running status on the
+cluster
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* Introduce an API `cluster/{int-id}/unmanage` for triggering an un-manage
+cluster flow
+
+==== Notifications/Monitoring impact:
+
+* A flow to archive the cluster specific graphite data
+
+* A flow to remove the grafana alerts dashboards for the cluster and its
+dependent entities
+
+* Raise an alert once cluster got un-managed with details like where to look
+for old graphite data etc
+
+==== Tendrl/common impact:
+
+* A flow un-manage cluster to be targeted at tendrl server node
+
+==== Tendrl/node_agent impact:
+
+None
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl Dashboard impact:
+
+* Following changes required in UI dashboards based on UX designs mentioned at
+https://redhat.invisionapp.com/share/8QCOEVEY9
+
+** Add an option namely `Unmanage` under kebab menu for each successfully
+imported and managed cluster
+
+** Add a dialog box which opens up on click event of `Unmanage` option from
+kebab menu of the cluster. This dialog box is for confirmation from user to
+start un-manage flow for the cluster
+
+===== Workflow
+
+* User clicks the `Unmanage` option from the kebab menu for a managed cluster
+
+* The click event triggers a dialog box with appropriate message. A sample
+message is available at
+https://redhat.invisionapp.com/share/8QCOEVEY9#/screens/273239640
+
+* There are 3 possible actions on this dialog
+
+** `Close` icon to close the dialog and no action performed for un-managing the
+cluster. User would be directed back to clusters list page
+
+** `Cancel` button to close the dialog and no action performed for un-managing the
+cluster. User would be directed back to clusters list page
+
+** `Unmanage` button to start the un-manage cluster task in backend. A message
+with task details gets displayed on dialog box. Sample message available at
+https://redhat.invisionapp.com/share/8QCOEVEY9#/screens/273239844
+
+** This final message after submission of the task for un-managing cluster would
+also provide a button to view the task details. A button `View Task Progress` is
+available for the same. User can opt to close this dialog and later user context
+menus to check the task updates
+
+** Once a cluster is being moved to un-managed state, the changes in properties
```

Comment:
Is this after cluster moved to un-manage state or just after the task is created and in-progress?

Reply:
Its while un-manage in progress. @a2batic can you ack this?

Reply:
@shtripat @nthomas-redhat, its when the task for unmanage has been submitted from UI and API has acknowledged the submission of task.